tools/topology/topology2/platform/intel/speaker-echo-ref.conf

```diff
@@ -0,0 +1,58 @@
+Define {
+	ECHO_REF_STREAM_NAME	"Echo reference"
+}
+
+Object.Pipeline {
+	passthrough-capture.2 {
+		index	$ECHO_REF_HOST_PIPELINE_ID
+
+		Object.Widget.copier.1 {
+			stream_name	$ECHO_REF_STREAM_NAME
+		}
+	}
+
+	dai-copier-be.2 {
+		index	$ECHO_REF_DAI_PIPELINE_ID
+
+		Object.Widget.copier."1" {
+			dai_index	1
+			dai_type	"SSP"
+			type		dai_out
+			copier_type	"SSP"
+			stream_name	$SPEAKER_CODEC_NAME
+			node_type	$I2S_LINK_INPUT_CLASS
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+		}
+	}
+}
+
+Object.PCM {
+	pcm."2" {
```

Comment:
BTW @yongzhi1 if you want to give it a try. you can start your object instances from 1 for the pcm and routes in this conf file with this PR https://github.com/alsa-project/alsa-utils/pull/188. The compiler will take care of avoiding clashes with the top-level file objects

Reply:
Lets do this as part of the topology2 feature cleanup. @jsarha fyi.